ios/RCCommonFunctionality.m

```diff
@@ -145,23 +177,29 @@ + (void)purchasePackage:(NSString *)packageIdentifier offering:(NSString *)offer
             completion(nil, [self payloadForError:error withExtraPayload:@{@"userCancelled": @(userCancelled)}]);
         } else {
             completion(@{
-                @"purchaserInfo": purchaserInfo.dictionary,
-                @"productIdentifier": transaction.payment.productIdentifier
+                    @"purchaserInfo": purchaserInfo.dictionary,
+                    @"productIdentifier": transaction.payment.productIdentifier
             }, nil);
         }
     };
 
     [RCPurchases.sharedPurchases offeringsWithCompletionBlock:^(RCOfferings *offerings, NSError *error) {
         RCPackage *aPackage = [[offerings offeringWithIdentifier:offeringIdentifier] packageWithIdentifier:packageIdentifier];
         if (aPackage) {
-            [RCPurchases.sharedPurchases purchasePackage:aPackage withCompletionBlock:completionBlock];
+            if (discountTimestamp) {
+                if (@available(iOS 12.2, *)) {
+                    SKPaymentDiscount *discount = self.discounts[discountTimestamp];
+                    if (discount) {
+                        [RCPurchases.sharedPurchases purchasePackage:aPackage withDiscount:discount completionBlock:completionBlock];
```

Comment:
since this logic is duplicated, maybe it can be extracted, i.e.: 
```
- (nullable SKPaymentDiscount *)discountIfAvailableWithTimestamp:(nullable NSString *)discountTimestamp
```
and then
```
nullable SKPaymentDiscount *discount = [self discountIfAvailableWithTimestamp:discountTimestamp];
```
would save you some duplicated logic and nested ifs

Reply:
I ended up not doing this because it was complicating the logic if I wanted to make sure that for < 12.2 the purchase without a discount would be performed. Take a look at the new code, maybe you come up with a solution